pinot-core/src/main/java/org/apache/pinot/core/util/trace/ContinuousJfrStarter.java

```diff
@@ -0,0 +1,164 @@
+  /// than 1% overhead.
+  public static final String CONFIGURATION = "configuration";
+  /// Default value for the configuration key.
+  public static final String DEFAULT_CONFIGURATION = "default";
+  /// Key that controls the name of the recording.
+  ///
+  /// This is used to identify the recording in the JFR UI.
+  /// The default value is 'pinot-continuous'.
+  public static final String NAME = "name";
+  /// Default value for the name key.
+  public static final String DEFAULT_NAME = "pinot-continuous";
+
+
+  /// Key that controls whether to dump the recording on exit.
+  public static final String DUMP_ON_EXIT = "dumpOnExit";
+  /// Default value for the dumpOnExit key.
+  ///
+  /// If true, the recording will be dumped to a file when the JVM exits.
+  public static final boolean DEFAULT_DUMP_ON_EXIT = true;
+  /// Key that controls the directory to store the recordings when dumped on exit.
+  ///
+  /// The default value is the current working directory.
+  /// The filename will be 'recording-<timestamp>.jfr'.
+  public static final String DIRECTORY = "directory";
+
+  /// Key that controls whether to buffer the recording to disk.
+  /// If false, the recording will only be kept in memory.
+  /// If true, the recording will be written to disk periodically. This may affect performance but increases the window
+  /// of time that can be recorded.
+  public static final String TO_DISK = "toDisk";
+  /// Default value for the toDisk key.
+  public static final boolean DEFAULT_TO_DISK = true;
+  /// Key that controls the maximum size of the recording file.
+  /// Once the file reaches this size, older events will be discarded.
+  /// If both maxSize and maxAge are set, the recording will be discarded when either condition is met.
+  ///
+  /// This is only used if toDisk is true.
+  /// The value is in bytes.
+  /// The default value is 200MB.
+  ///
+  /// @see #MAX_AGE
+  public static final String MAX_SIZE = "maxSize";
+  public static final int DEFAULT_MAX_SIZE = 200 * 1024 * 1024;
+  /// Key that controls the maximum age of the recording file.
+  /// Once the file reaches this age, older events will be discarded.
+  /// If both maxSize and maxAge are set, the recording will be discarded when either condition is met.
+  ///
+  /// This is only used if toDisk is true.
+  /// The value is a duration string, as defined by the Duration class.
+  /// The default value is 1 day (P1D).
+  ///
+  /// @see #MAX_SIZE
+  public static final String MAX_AGE = "maxAge";
+  public static final String DEFAULT_MAX_AGE = "P1D";
+
+  /// A flag to track whether the JFR recording has been started.
+  /// This is specially useful for testing and quickstarts, where servers, brokers and other components are executed
+  /// in the same JVM.
+  private static boolean _started = false;
+
+  private ContinuousJfrStarter() {
+  }
+
+  public synchronized static void init(PinotConfiguration config) {
```

Comment:
Should we call this from minions as well for consistency?

Reply:
Done